docs/project-file-structure.md

```diff
@@ -64,30 +91,139 @@ to organize their structure
 
 Resources, scenes, scripts etc. used exclusively within that scene should be stored in the scenes subdirectory
 
+A scene and its script should always be together within the directory i.e. no /scripts/ and /scenes/ directories should exist within
+the project somewhere.
+
 Example:
 ```
 root/
 |- src/
-    |- Theme/
+    |- theme/
     |   |- ButtonNormal.tres
     |   |- ButtonHover.tres
     |   |- ButtonDisabled.tres
-    |- Scripts/
+    |- utility/
     |   |- Math.gd
-    |- Menu/
+    |- menu/
     |   |- MenuManager.tscn
     |   |- MenuManager.gd
     |   |- MainMenu/
     |   |   |- MainMenu.tscn
     |   |   |- MainMenu.gd
     |   |   |- MainMenuFont.ttf
     |   |   |- MainMenuLabel.tres
-    |   |- SettingsMenu/
+    |   |- settings_menu/
     |   |   |- Settings.tscn
     |   |   |- Settings.gd
     |   |   |- Resolution/
     |   |   |   |- ResolutionSelector.gd
-    |   |   |- Volume/
+    |   |   |- volume/
     |   |   |   |- VolumeGrid.tscn
     |   |   |   |- VolumeGrid.gd
 ```
+
+#### Naming Convention
+
+Directory names should be snake_case.
+```
+root/src/game_session/
+```
+
+File names should be PascalCase.
+```
+root/src/game_session/Map.tscn
+```
```

Comment:
Subdirectories within Godot's `root/src/` folder should be in Pascal case to better reflect the file (and class) names they encompass

````suggestion
Subdirectories for Godot files should be PascalCase.
```
root/src/GameSession/
```

File names should be PascalCase.
```
root/src/GameSession/Map.tscn
```
````

Reply:
I personally think we should keep directory names consistent throughout the project. Having PascalCase for godot but not cpp and assets/data seems counterintuitive to me

Reply:
We should definitely keep everything PascalCase inside the game/ directory, the C++ I know has its own conventions and I'd be fine with keeping them separated, but when we're working with the game we should have one set of rules, definitely.

Reply:
We are using PascalCase for godot directories and snake_case for other directories.

The reason is because it makes a clear distinction between directories that will get exported in the executable generated by godot and the directories that are not.